scaffolding-ruby/doc/reference.md

```diff
@@ -0,0 +1,473 @@
+* If your app's root directory contains a `.git/` subdirectory, then Git-related Habitat packages will be injected into your Plan's `pkg_build_deps` array to better support installing your app while honoring the `.gitignore` file.
+
+###  Specifying Run Dependencies in Your Plan
+
+Generally speaking, your app needs additional Habitat packages present at runtime for one of two reasons:
+
+1. Your app expects certain programs present which it may shell out  or bind to in some way.
+2. Your `Gemfile` has gem dependencies with native extensions, requiring additional linked libraries.
+
+In both cases, this means your app needs additional Habitat package installed at runtime. For this, we use the `pkg_deps` array in your Plan.
+
+For this example, let's suppose you are using the [rbnacl][] gem for bindings to the [libsodium][] library. Additionally, your app resizes images to generate thumbnails by using the `mogrify` program from the [ImageMagick][] software. To add both of these packages, update your `pkg_deps` line to the following below:
+
+```sh
+pkg_name=my_app
+pkg_origin=acmecorp
+pkg_version=0.1.0
+ # ...
+pkg_scaffolding=core/scaffolding-ruby
+pkg_deps=(core/imagemagick core/libsodium)
+```
+
+###  Specifying Build Dependencies in Your Plan
+
+Finally, your app may require additional software present in order to **build** your app that is not present by default.
+
+For this example, let's suppose you are using the [Helix][] project to write some Ruby classes in Rust. For this you need the Rust compiler present, but only at build time. To add the Rust compiler package, update your `pkg_build_deps` line to the following below:
+
+```sh
+pkg_name=my_app
+pkg_origin=acmecorp
+pkg_version=0.1.0
+ # ...
+pkg_scaffolding=core/scaffolding-ruby
+pkg_build_deps=(core/rust)
+```
+
+## Ruby Version
+
+### Selecting a Version of Ruby
+
+By default the latest version of the [MRI][]-based [`core/ruby`][] package will be injected into your Plan's `pkg_deps` array. To specify a non-default version of Ruby, there are two locations you can do this:
+
+1. Use the [`ruby`][gemfile_ruby] keyword in your app's `Gemfile`
+2. Set the `scaffolding_ruby_pkg` variable in your Plan with a valid Habitat package identifier corresponding to a package with a `ruby` program
+
+A set Plan variable will win over setting a version in your `Gemfile`, however it is recommended to the `Gemfile` strategy first as this is portable across other Ruby app build and deployment solutions.
+
+#### Specifying a Ruby Version in Your Gemfile
+
+You can use the [`ruby`][gemfile_ruby] keyword in your app's `Gemfile` to specify a version of Ruby. For example:
+
+```ruby
+source "https://rubygems.org"
+
+ruby "2.4.1"
+
+ # ...
+```
+
+The value of this keyword will be used to determine the version of the Habitat [`core/ruby`][] package. For example, the `Gemfile` above would result in `core/ruby/2.4.1` being injected into your Plan's `pkg_deps` array.
```

Comment:
I added a note here to this section with a link to the Depot page. At the moment, there is very simple logic to match-and-calculate the Ruby package which we can improve as we move along. Especially if that involves mapping Ruby 2.1.x versions to the `core/ruby21` package, etc. It's a good point and something I forgot about in the heat of working on this.